src/main/java/uk/ac/cam/cl/dtg/segue/search/IsaacSearchQueryBuilder.java

```diff
@@ -0,0 +1,198 @@
+        this.query = buildBaseSearchQuery(new BooleanMatchInstruction());
+    }
+
+    /*
+    The base query excludes no-filter content, content marked deprecated, unpublished content (depending on
+    config properties) and content tagged with "regression-test" (depending on config properties).
+     */
+    final BooleanMatchInstruction buildBaseSearchQuery(final BooleanMatchInstruction query) {
+        // Exclude unpublished content (based on config)
+        if (this.includeOnlyPublishedContent) {
+            query.mustNot(new MustMatchInstruction(Constants.PUBLISHED_FIELDNAME, "false"));
+        }
+
+        // Exclude regression test content (based on config)
+        if (this.excludeRegressionTestContent) {
+            query.mustNot(new MustMatchInstruction(Constants.TAGS_FIELDNAME, "false"));
+        }
+
+        // Exclude "no-filter" content
+        query.mustNot(new MustMatchInstruction(Constants.TAGS_FIELDNAME, HIDE_FROM_FILTER_TAG));
+
+        // Exclude deprecated content
+        query.mustNot(new MustMatchInstruction(Constants.DEPRECATED_FIELDNAME, "true"));
+
+        return query;
+    }
+
+    /**
+     * Sets the types of content to match. If left unspecified, the built query will match all content types.
+     *
+     * @param contentTypes A list of content types to match.
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchQueryBuilder matchContentTypes(final Set<String> contentTypes) {
+        this.matchContentTypes = contentTypes;
+        return this;
+    }
+
+    /**
+     * Sets the types of content to prioritise. Expected to be a subset of the matched content types.
+     * If left unspecified, the built query will not prioritise any content types.
+     *
+     * @param contentTypes A list of content types to match.
+     * @return This IsaacSearchQueryBuilder, to allow chained operations.
+     */
+    public IsaacSearchQueryBuilder prioritiseContentTypes(final Set<String> contentTypes) {
+        this.priorityContentTypes = contentTypes;
+        return this;
+    }
+
+    public IsaacSearchQueryBuilder matchFields(final Set<String> fields) {
+        this.matchFields = fields;
+        return this;
+    }
+
+    public IsaacSearchQueryBuilder prioritiseFields(final Set<String> fields) {
+        this.priorityMatchFields = fields;
+        return this;
+    }
+
+
+
+    public BooleanMatchInstruction build() {
+        String nestedFieldConnector = searchProvider.getNestedFieldConnector();
+
+        int minimumMatchInstructions = 1;
+
+        List<String> contentTypes = Optional.ofNullable(this.matchContentTypes)
+                .orElse(Collections.emptySet())
+                .stream()
+                .filter(SITE_WIDE_SEARCH_VALID_DOC_TYPES::contains)
+                .collect(Collectors.toList());
+
+        if (contentTypes.isEmpty()) {
+            contentTypes = Lists.newArrayList(SITE_WIDE_SEARCH_VALID_DOC_TYPES);
```

Comment:
The doctypes which are valid for site wide search might not be valid for other search queries.